core/domain/question_services_test.py

```diff
@@ -974,3 +974,115 @@ def test_migrate_question_state_from_v32_to_v33(self):
+            },
+            'recorded_voiceovers': {
+                'voiceovers_mapping': {}
+            },
+            'written_translations': {
+                'translations_mapping': {
+                    'explanation': {}
+                }
+            },
+            'interaction': {
+                'answer_groups': [answer_group],
+                'confirmed_unclassified_answers': [],
+                'customization_args': {
+                    'choices': {
+                        'value': ''
+                    },
+                    'showChoicesInShuffledOrder': {
+                        'value': True
+                    }
+                },
+                'default_outcome': {
+                    'dest': None,
+                    'feedback': {
+                        'content_id': 'feedback_1',
+                        'html': 'Correct Answer'
+                    },
+                    'param_changes': [],
+                    'refresher_exploration_id': None,
+                    'labelled_as_correct': True,
+                    'missing_prerequisite_skill_id': None
+                },
+                'hints': [{
+                    'hint_content': {
+                        'content_id': 'hint_1',
+                        'html': 'Hint 1'
+                    }
+                }],
+                'solution': {},
+                'id': 'MultipleChoiceInput'
+            },
+            'param_changes': [],
+            'solicit_answer_details': False,
+            'classifier_model_id': None
+        }
+        expected_feeedback_html = (
+            '<p>Feedback</p><oppia-noninteractive-' +
+            'math math_content-with-value="{&amp;quot' +
+            ';raw_latex&amp;quot;: &amp;quot;+,-,-,+' +
+            '&amp;quot;, &amp;quot;svg_filename&amp;' +
+            'quot;: &amp;quot;&amp;quot;}"></oppia' +
+            '-noninteractive-math>')
+
+        question_model = (
+            question_models.QuestionModel(
+                id='question_id',
+                question_state_data=question_state_dict,
+                language_code='en',
+                version=0,
+                linked_skill_ids=['skill_id'],
+                question_state_data_schema_version=33))
+        commit_cmd = (
+            question_domain.QuestionChange({
+                'cmd': question_domain.CMD_CREATE_NEW
+            }))
+        commit_cmd_dicts = [commit_cmd.to_dict()]
+        question_model.commit(
+            'user_id_admin', 'question model created', commit_cmd_dicts)
+
+        current_schema_version_swap = self.swap(
+            feconf, 'CURRENT_STATE_SCHEMA_VERSION', 34)
+
+        with current_schema_version_swap:
+            question = question_fetchers.get_question_from_model(question_model)
+
+        self.assertEqual(question.question_state_data_schema_version, 34)
+
+        migrated_answer_group = (
+            question.question_state_data.interaction.answer_groups[0].to_dict())
```

Comment:
ANy specific reason for conversion to dict?

Reply:
changed